resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala

```diff
@@ -152,12 +152,6 @@ private object ResourceRequestHelper extends Logging {
       return
     }
 
-    if (!isYarnResourceTypesAvailable()) {
-      logWarning("Ignoring custom resource requests because " +
-          "the version of YARN does not support it!")
-      return
-    }
-
     val resInfoClass = Utils.classForName(RESOURCE_INFO_CLASS)
```

Comment:
Other uses related to reflection in `ResourceRequestHelper` will be cleared in SPARK-43216